cli/lang-generator.js

```diff
@@ -0,0 +1,91 @@
+const fs = require('fs');
+const path = require('path');
+const sergeDirectories = require('../core.serge.json');
+
+const outputRoot = path.join(__dirname, '../generated');
+const outputPath = path.join(outputRoot, 'lang');
+
+function alphaSort(a, b) {
+	a = a.toLowerCase();
+	b = b.toLowerCase();
+	if (a < b) {
+		return -1;
+	} else if (a > b) {
+		return 1;
+	}
+	return 0;
+}
+
+function cleanDir(dirPath) {
+	if (!fs.existsSync(dirPath)) {
+		return;
+	}
+	const files = fs.readdirSync(dirPath);
+	files.forEach((file) => {
+		const currentPath = path.join(dirPath, file);
+		if (fs.lstatSync(currentPath).isDirectory()) {
+			cleanDir(currentPath);
+		} else {
+			fs.unlinkSync(currentPath);
+		}
+	});
+
+	fs.rmdirSync(dirPath);
+
+}
+function appendComponent(sergeComponent) {
+	const directory = `${sergeComponent.source_dir}/`;
+	const filenames = fs.readdirSync(directory);
+
+	if (!filenames) {
+		throw 'dir is empty';
+	}
+
+	filenames.forEach((filename) => {
+		const filepath = directory + filename;
+		const lang = path.basename(filepath, '.json');
+
+		const terms = {};
+		const data = fs.readFileSync(filepath);
+		if (data) {
+			const currentLangTerms = JSON.parse(data);
+			Object.keys(currentLangTerms)
```

Comment:
Somewhat of an aside, how are untranslated terms identified and tracked?  There might be some value in identifying such terms and logging them.  That would likely be a separate util if we needed/wanted.

Reply:
To know if a term hasn't been translated.
New terms would only be in the `en.json` file.
For updated terms, serge does a diff on the previous `en.json` file to see if anything has changed.